vscode-trace-common/src/client/tsp-client-provider-impl.ts

```diff
@@ -10,13 +10,21 @@ export class TspClientProvider implements ITspClientProvider {
     private _traceManager: TraceManager;
     private _experimentManager: ExperimentManager;
     private _listeners: ((tspClient: TspClient) => void)[] = [];
+    private _initialized = false;
 
     constructor(
         private _url: string,
         private _signalHandler: VsCodeMessageManager | undefined
     ) {
         this.updateClients();
-        RestClient.addConnectionStatusListener(status => this._signalHandler?.notifyConnection(status));
+
+        RestClient.addConnectionStatusListener(status => {
+            // Ignore the first update that is sent when calling addConnectionStatusListener
+            if (this._initialized) {
+                this._signalHandler?.notifyConnection(status);
+            }
+        });
+        this._initialized = true;
```

Comment:
Similar pattern to this: https://github.com/eclipse-cdt-cloud/vscode-trace-extension/pull/216/commits/f565c231cd9fd15aa0ab6932ec83a4a2dc96d651.

I wonder if there's a cleaner way to do this.

Reply:
I'm not sure what you mean. What do you suggest to make it cleaner? Please let me know

Reply:
Ok, I think I understand how to change it. I'll update the PR (including a rebase).